### ISSUE TITLE
Way to hide/control dependency output in yarn upgrade/install?

### ISSUE THREAD

**Issue by henderea** [yarnpkg/yarn#7148]
**Do you want to request a *feature* or report a *bug*?**

Feature

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce.**

When running `yarn upgrade` or `yarn install`, it finishes up by printing out the dependencies.  I can hide this with `--silent`, but that also hides other stuff like the nice progress view and the execution time.  On a project with a lot of dependencies, this can mean a lot of junk getting printed out on every `yarn upgrade`.

**What is the expected behavior?**
There should be a way to hide the dependency information printed out after a `yarn upgrade` or `yarn install`.

Another option would be to only show what actually _changed_.  Printing out the full dependency list when the upgrade doesn't even find any new versions of anything is kind of useless.  If you _do_ show the full list, then at least indicate the actual changes somehow, like the Ruby `bundle update` command.

A third option that is a bit off topic for the issue but would still cover my needs would be to have a flag that forces the full progress view in silent mode, or a way to hide specific log levels like info.  Maybe a `--advanced-progress` or `--no-log info` flag would cover my use case.

**Please mention your node.js, yarn and operating system version.**
macOS 10.14.3
node 11.12.0
yarn 1.15.2

**Comment by janosh** [comment]
I completely agree. It's great that `yarn` tells me that the update involved 731 dependencies but there's no need to then go on printing _every single one of them_ to the console.

```sh
[4/4] 🔨  Rebuilding all packages...
success Saved lockfile.
success Saved 731 new dependencies.
info Direct dependencies
├─ contentful-cli@0.36.0
├─ eslint@6.1.0
├─ gatsby-cli@2.7.28
├─ gatsby-dev-cli@2.5.7
├─ jest-cli@24.8.0
├─ netlify-cli@2.12.0
├─ tslint@5.18.0
└─ typescript@3.5.3
info All dependencies
├─ @babel/generator@7.5.5
├─ @babel/helper-function-name@7.1.0
├─ @babel/helper-get-function-arity@7.0.0
├─ @babel/helper-split-export-declaration@7.4.4
├─ @babel/helpers@7.5.5
├─ @babel/highlight@7.5.0
├─ @babel/parser@7.5.5
├─ @babel/plugin-syntax-object-rest-spread@7.2.0
├─ @babel/template@7.4.4
├─ @babel/traverse@7.5.5
...
```
